app/db/repositories/location_repo.py

```
@@ -0,0 +1,105 @@
+                return new_location
+            except SQLAlchemyError as e:
+                # In a real app, log error e
+                await session.rollback()
+                return None
+
+    async def get_location_by_id(self, location_id: int) -> Optional[Location]:
+        async with self.session_factory() as session:
+            try:
+                result = await session.execute(select(Location).filter_by(id=location_id))
+                return result.scalar_one_or_none()
+            except SQLAlchemyError as e:
```

Ruff (0.11.9)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)